File: examples/R/random_assignment.R
# Simple random assignment helper in R.
#
# Usage:
#   Rscript examples/R/random_assignment.R

random_assign <- function(ids, groups) {
  ids <- sample(ids)  # shuffle
  n_groups <- length(groups)
  assigned_groups <- groups[((seq_along(ids) - 1) %% n_groups) + 1]
  data.frame(id = ids, group = assigned_groups, stringsAsFactors = FALSE)
}

main <- function() {
  cat("Basit Rastgele Atama Aracı (R)\n")
  raw_ids <- readline("Katılımcı ID'lerini virgülle ayırarak girin (örn: S1,S2,S3): ")
  ids <- trimws(unlist(strsplit(raw_ids, ",")))
  ids <- ids[ids != ""]
  if (length(ids) == 0) {
    cat("Hiç katılımcı girilmedi.\n")
    return(invisible(NULL))
  }

  raw_groups <- readline("Grupları virgülle ayırarak girin (örn: A,B): ")
  groups <- trimws(unlist(strsplit(raw_groups, ",")))
  groups <- groups[groups != ""]
  if (length(groups) < 2) {
    cat("En az iki grup girilmelidir.\n")
    return(invisible(NULL))
  }

  res <- random_assign(ids, groups)
  cat("\nAtama Sonuçları:\n")
  print(res, row.names = FALSE)
}

if (identical(environment(), globalenv())) {
  main()
}

File: examples/python/random_assignment.py
"""Simple random assignment helper for experimental projects.

Given a list of participant IDs, assign them randomly to groups
(A/B or more groups).

Usage example:
    python examples/python/random_assignment.py
"""

from __future__ import annotations

import random
from typing import Dict, List


def random_assign(ids: List[str], groups: List[str]) -> Dict[str, str]:
    shuffled = ids[:]
    random.shuffle(shuffled)
    assignments: Dict[str, str] = {}
    for i, pid in enumerate(shuffled):
        assignments[pid] = groups[i % len(groups)]
    return assignments


def main() -> None:
    print("Basit Rastgele Atama Aracı")
    raw_ids = input("Katılımcı ID'lerini virgülle ayırarak girin (örn: S1,S2,S3): ")
    ids = [s.strip() for s in raw_ids.split(",") if s.strip()]
    if not ids:
        print("Hiç katılımcı girilmedi.")
        return

    raw_groups = input("Grupları virgülle ayırarak girin (örn: A,B): ")
    groups = [g.strip() for g in raw_groups.split(",") if g.strip()]
    if len(groups) < 2:
        print("En az iki grup girilmelidir.")
        return

    assignments = random_assign(ids, groups)
    print("\nAtama Sonuçları:")
    for pid in sorted(assignments.keys()):
        print(f"- {pid} -> {assignments[pid]}")


if __name__ == "__main__":
    main()
